File: 2716. Minimize String Length.py
#PYTHON
#* 2716. Minimize String Length
#* LEARN => COUNTER == TALLY RUBY
s = "aaabc"
from collections import Counter

def minimized_string_length(s):
    return len(Counter(s))


print(minimized_string_length(s))  # Output: 3
print(Counter(s))

File: 2716. Minimize String Length.cs
// 2716. Minimize String Length
// s = "aaabc"
//SOL
public class Solution {
    public int MinimizedStringLength(string s) {
        Dictionary<char, int> charCount = new Dictionary<char, int>();
        
        foreach (char c in s) {
            if (charCount.ContainsKey(c)) {
                charCount[c]++;
            } else {
                charCount[c] = 1;
            }
        }
        return charCount.Count;
    }
}

